### PR TITLE
Chromebook Version 89 Serial Port Fix

### PR DESCRIPTION
**Pull request by epeach**
Context:
SerialPort is how we connect from the browser to the USB. Since the SerialPort was not previously provided in the browser and we exposed it in the Maker App. Later in our code, we use whether or not the SerialPort is exposed as a way to determine whether we were in the Maker App or Chrome App. However, the latest version of Chrome (89) exposes WebSerial as a SerialPort on the browser. This is excellent news for our future work of deprecating the Chrome App! Unfortunately, it meant we could no longer use whether or not the SerialPort was exposed as a way to determine whether or not we were in the Chrome App (and subsequently do Chrome App specific things) or in the Maker App.

The Change:
We already have a function called isChromeOS, so I replaced the logic on isNodeSerialAvailable with !isChromeOS. I discussed this with Maddie to figure out if it was logically sound. We agreed it was - summary: if we are in ChromeOS, we want to behave as if nodeSerial is not available. If we are not in ChromeOS, we are either in Maker App (node serial is available) or a regular browser (doesn't work anyways).
After doing this, I realized that all of our tests for the CircuitPlaygroundBoard are set up to run as if they were on ChromeOS. So I had to stub tests to believe they are in ChromeOS to get the tests up and running. This means there is going to be future work when we deprecate the Chrome App, but I don't want to block this PR on that.

History of the Change:
Today happened to be a day that we couldn't spin up adhocs! After realizing that was off the plate, we explored using an EC2 from a Windows machine to run the code and test on the Chromebook. However, that ran into the same funkiness we had with trying to ngrok to the Chromebook (it doesn't seem to recognize when the Chrome App is installed?). In the end, after the DTP had gone out for they day, Dotd froze the deploy pipeline and allowed us to merge to staging to test. (https://github.com/code-dot-org/code-dot-org/pull/39686) But then it broke the tests so we reverted (https://github.com/code-dot-org/code-dot-org/pull/39690), commented out the tests, and re-merged (https://github.com/code-dot-org/code-dot-org/pull/39691). Then we realized there was one more test that needed to be commented out and decided to open a new PR to comment that out (https://github.com/code-dot-org/code-dot-org/pull/39692), instead of reverting/unreverting once more. Once we found out that works, I reverted the temp test fix (https://github.com/code-dot-org/code-dot-org/pull/39695) and the original revert chain (https://github.com/code-dot-org/code-dot-org/pull/39696). Then I ported all the changes to this branch. Whew!

## Testing story
Tested locally on Chromebook. Tested locally on Linux/MakerApp. Tested locally on Mac/MakerApp.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js [hunk 1]
```diff
@@ -58,10 +62,20 @@ export function itMakesCircuitPlaygroundComponentsAvailable(BoardClient) {
         },
         addCustomMarshalObject: sinon.spy()
       };
+      // Opportunity to stub any needed to test a board
```
Comment (ajpal) [inline apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js]: nit: "anything"? or am I misreading?

Comment (epeach) [reply apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js]: Yup, typo . . . fixing . . .

File: apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js [hunk 2]
```diff
@@ -316,16 +331,33 @@ describe('CircuitPlaygroundBoard', () => {
     CircuitPlaygroundBoard.makePlaygroundTransport.restore();
     EventEmitter.prototype.once.restore();
     ChromeSerialPort.stub.reset();
+    circuitPlaygroundBoardTeardown();
   });
 
+  function circuitPlaygroundBoardSetup() {
+    spy = sinon.stub(navigator, 'userAgent').value('CrOS');
```
Comment (ajpal) [inline apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js]: nit: maybe leave a comment that CrOS == chrome OS?

**Review by ajpal — APPROVED** [review]

File: apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js (hunk 2 above)
Comment (maddiedierker) [inline apps/test/unit/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoardTest.js]: nit: i think `spy` is too generic here. maybe `userAgentSpy`?

File: apps/test/unit/lib/kits/maker/portScanningTest.js [hunk 3]
```diff
@@ -15,12 +15,19 @@ import {
   findPortWithViableDevice,
   getPreferredPort
 } from '@cdo/apps/lib/kits/maker/portScanning';
+import sinon from 'sinon';
 
 describe('maker/portScanning.js', function() {
+  let spy;
   describe(`findPortWithViableDevice()`, () => {
+    beforeEach(() => {
+      spy = sinon.stub(navigator, 'userAgent').value('CrOS');
```
Comment (maddiedierker) [inline apps/test/unit/lib/kits/maker/portScanningTest.js]: same `userAgentSpy` nit here 😄

**Review by maddiedierker — APPROVED** [review]
left some tiny naming nits that shouldn't block this PR. great find and fix!

bmo

**Pull request merged**